src/protocol.rs

```diff
@@ -289,16 +286,20 @@ impl Participant {
     ) -> Result<SignatureShare> {
         info!("Issuing signature with presign record.");
 
-        let presign_record: PresignRecord =
-            self.main_storage
-                .retrieve(PresignStorageType::Record, presign_identifier, self.id)?;
+        let presign_record: PresignRecord = self.main_storage.retrieve(
+            PersistentStorageType::PresignRecord,
+            presign_identifier,
+            self.id,
+        )?;
         let (r, s) = presign_record.sign(digest)?;
         let ret = SignatureShare { r: Some(r), s };
 
         // Clear the presign record after being used once
-        let _ =
-            self.main_storage
-                .delete(PresignStorageType::Record, presign_identifier, self.id)?;
+        let _ = self.main_storage.delete(
+            PersistentStorageType::PresignRecord,
+            presign_identifier,
+            self.id,
+        )?;
```

Comment:
thought: this is not really the point of this PR but I would be inclined to delete this even if `sign` fails on L294, e.g. immediately after `retrieve`.